File: music-infrastructure/src/main/java/com/music/sale/persistence/user/entity/UserEntity.java
package com.music.sale.persistence.user.entity;

import com.music.sale.persistence.common.BaseEntity;
import com.music.sale.persistence.user.enums.UserRole;
import com.music.sale.persistence.user.enums.UserStatus;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Setter
@Builder
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Table(
    name = "users",
    indexes = {
      @Index(name = "idx_user_ci", columnList = "ci"), // 본인인증으로 사람 찾기용
      @Index(name = "idx_user_email", columnList = "email") // 이메일 검색용
    })
public class UserEntity extends BaseEntity {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "user_id")
  private Long id;

  @Column(length = 50)
  private String nickname;

  @Column(length = 300)
  private String profileImageUrl;

  @Column(length = 100)
  private String email;

  @Enumerated(EnumType.STRING)
  @Column(nullable = false)
  private UserRole role;

  @Enumerated(EnumType.STRING)
  @Column(nullable = false)
  private UserStatus status;

  @Column(unique = true, length = 100)
  private String ci;

  @Column(length = 50)
  private String realName;

  @Column(length = 20)
  private String phoneNumber;

  @Column(length = 8)
  private String birthDate;

  @Column(nullable = false)
  private boolean isVerified;
}

File: bin/main/com/music/sale/adapter/persistence/user/entity/UserEntity.kt
package com.music.sale.adapter.persistence.user.entity

import com.music.sale.adapter.persistence.common.BaseEntity
import com.music.sale.domain.user.User
import com.music.sale.domain.user.enum.UserRole
import jakarta.persistence.*

@Entity
@Table(name = "users")
class UserEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(nullable = false, unique = true)
    val email: String,

    @Column(nullable = false)
    var name: String,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    var role: UserRole

) : BaseEntity() {

    fun toDomain(): User {
        return User(
            id = id!!,
            email = User.Email(email),
            name = User.Name(name),
            role = role
        )
    }

    companion object {
        fun fromDomain(user: User): UserEntity {
            return UserEntity(
                id = user.id,
                email = user.email?.value ?: "",
                name = user.getName()?.value ?: "",
                role = user.getRole() ?: UserRole.USER
            )
        }
    }
} 